pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
How about just storing the list of InterfaceInfo, and we always pick the first one. The plugin can store IPv4 or IPv6 as it pleases as golang's net.IP will store either.
```go
type InterfaceInfo struct {
    Iface net.Interface
    IPAddrs []net.IP
}

type NetworkStatus struct {
     NetworkInterfaces []net.InterfaceInfo
}

// kube code.. null checks excluded
result.ip = netStatus.NetworkInterfaces[0].IPAddrs[0].String()
```

Alternately, we can store a primary IPAddress at the top (like docker's NetworkSettings.IPAddress):
```go
type InterfaceInfo struct {
    Iface net.Interface
    IPAddrs []net.IP
}

type NetworkStatus struct {
    IPAddress net.IP  // Primary ipv4/ipv6 address of the pod
    NetworkInterfaces []net.InterfaceInfo
}
```